spresense-ethernet-examples/Ethernet-v2.0.0-SPRESENSE/Ethernet/src/Dhcp.cpp

```
@@ -0,0 +1,433 @@
+	memset(_dhcpMacAddr, 0, 6);
+	reset_DHCP_lease();
+
+	memcpy((void*)_dhcpMacAddr, (void*)mac, 6);
+	_dhcp_state = STATE_DHCP_START;
+	return request_DHCP_lease();
+}
+
+void DhcpClass::reset_DHCP_lease()
+{
+	// zero out _dhcpSubnetMask, _dhcpGatewayIp, _dhcpLocalIp, _dhcpDhcpServerIp, _dhcpDnsServerIp
+	memset(_dhcpLocalIp, 0, 20);
```

Cppcheck (2.10-2)

[error] Buffer is accessed out of bounds

(bufferAccessOutOfBounds)